url58b12a0c74_tatianaslod_daooTP_tgz-pJ8-tp1_MainTp1J8/src/server/DaooTaskServer.java

```diff
@@ -0,0 +1,31 @@
+package server;
+
+import com.sun.istack.internal.NotNull;
+import daoo.ioc.Task;
+import daoo.ioc.TaskExecutor;
+import daoo.ioc.TaskServer;
+import ioc.Context;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class DaooTaskServer implements TaskServer {
+    @Override public void start(@NotNull TaskExecutor executor, int port) {
+        try {
+            ServerSocket serverSocket = new ServerSocket(port);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *serverSocket*. Execution paths that do not contain closure statements were detected. Ensure *serverSocket* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).